subprojects/core/src/main/java/org/gradle/execution/plan/WorkNodeExecutor.java

```diff
@@ -24,7 +24,7 @@ public boolean execute(Node node, ProjectExecutionServiceRegistry services) {
         if (!(node instanceof ActionNode)) {
             return false;
         }
-        ((ActionNode) node).run();
+        ((ActionNode) node).run(services);
```

Comment:
Can you clarify your comment? I don't see the pattern where nodes receice a service registry. `ProjectExecutionServices` is only used in one place: [LocalTaskNodeExecutor](https://github.com/gradle/gradle/blob/8fc8e0e0c569948c6cdecc390579b9301efea43c/subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNodeExecutor.java#L30-L49).

The node itself does not receive the service registry, it is used in the node execution.

Is your point here that I should pass in the service registry for the project instead of the `ProjectExecutionServiceRegistry`?

Reply:
Remove "'d". just wanted to say that it's a good pattern I think we could use everywhere, to expose the services available for execution to the node during, well, execution.

Reply:
👍